src/App.js

```diff
@@ -1,13 +1,51 @@
 import React from 'react';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component {
+  state = {
+    goods: [],
+  }
+
+  renderGoods = (callback) => {
+    callback().then(goods => this.setState({ goods }));
+  }
+
+  render() {
+    const { goods } = this.state;
+
+    return (
+      <div className="app">
+        <button
+          className="app__button"
+          type="submit"
+          onClick={() => this.renderGoods(getAll)}
+        >
+          Load All goods
+        </button>
+        <button
+          type="submit"
+          className="app__button"
+          onClick={() => this.renderGoods(get5First)}
+        >
+          Load 5 first goods
+        </button>
+        <button
+          className="app__button"
+          type="submit"
+          onClick={() => this.renderGoods(getRedGoods)}
+        >
+          Load red goods
+        </button>
+        {!goods.length || (
+          <GoodsList goods={goods} />
+        )}
```

Comment:
use && instead of || operator
```suggestion
        {goods.length > 0 && (
          <GoodsList goods={goods} />
        )}
```